pandas-stubs/core/groupby/generic.pyi

```diff
@@ -70,16 +84,18 @@ class SeriesGroupBy(GroupBy[Series[S1]], Generic[S1, ByT]):
     def aggregate(
         self,
         func: AggFuncTypeBase | None = ...,
+        /,
         *args,
```

Comment:
Before this overload, you could add this overload:
```python
    @overload
    def aggregate(
        self,
        func: Callable[[Series], S2],
        *args,
        engine: WindowingEngine = ...,
        engine_kwargs: WindowingEngineKwargs = ...,
        **kwargs,
    ) -> Series[S2]: ...
```
Then you know that if you start with a `Series` with a known type, then the return type would be inferred from the callable.  And it works with a lambda function, e.g.:
```python
    s = pd.Series([1, 2, 3, 4])
    q = s.groupby([1, 1, 2, 2]).agg(lambda x: x.astype(float).min())
```
In this case, `q` would have type `Series[float]`, which is what you want.

Reply:
<img width="780" alt="image" src="https://github.com/user-attachments/assets/7b506025-1b8f-4d68-9def-8a6b67ddcdc4" />
Even with the new overload mypy still complains (pyright does not). I think it does not recognize the lamda as returning S2 (float).

Reply:
I think that's because the type of `new_func` isn't clear.

But I think it would work if you did `check(assert_type(s.groupby([1,1,2,2]).agg(lambda x: x.astype(float).min()), "pd.Series[int]"), pd.Series, int)`

Because then it can know that `x` is a `Series[int]` and that the `lambda` becomes `Series[int]`

Can you try that?

Reply:
I tried that for the last push, see https://github.com/pandas-dev/pandas-stubs/blob/f9863d03c387dcca86da8f86675a105a5d837786/tests/test_series.py#L1167
It fails in all CI:
```
===========================================
Beginning: 'Run mypy on 'tests' (using the local stubs) and on the local stubs'
===========================================

tests/test_series.py:1167: error: Expression is of type "Series[Any]", not "Series[float]"  [assert-type]
Found 1 error in 1 file (checked 224 source files)
```